### PR TITLE
Use gitea forked macaron

### LINKED ISSUE #7670: Update go-macaron/cache and go-macaron/session
I would like to update the go-macaron/cache and go-macaron/session packages to use the latest version of go-redis client. The latest client has support for Redis cluster. I opened prs against the macaron repos for 2 months which have not been merged.

- https://github.com/go-macaron/cache/pull/12
- https://github.com/go-macaron/session/pull/28

How can gitea use my updated version of the macaron libraries?

### PR DESCRIPTION
**Pull request by tamalsaha**
Signed-off-by: Tamal Saha <tamal@appscode.com>

Fixes https://github.com/go-gitea/gitea/issues/7670

### CODE REVIEW

**Comment by tamalsaha** [comment]
Please note that `github.com/Unknwon/cae` uses capital `U` because the package itself imports with capital `U`. I have opened a pr https://github.com/unknwon/cae/pull/8

**Comment by lunny** [comment]
replaced #7930

File: integrations/git_helper_for_declarative_test.go [hunk 1]
```diff
@@ -19,8 +19,8 @@ import (
 	"code.gitea.io/gitea/modules/git"
 	"code.gitea.io/gitea/modules/setting"
 	"code.gitea.io/gitea/modules/ssh"
-	"github.com/Unknwon/com"
 	"github.com/stretchr/testify/assert"
```
Comment (lunny) [inline integrations/git_helper_for_declarative_test.go:23]: place a blank line before internal packages and external packages.

File: models/git_diff.go [hunk 2]
```diff
@@ -25,8 +25,8 @@ import (
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/process"
 	"code.gitea.io/gitea/modules/setting"
-	"github.com/Unknwon/com"
 	"github.com/sergi/go-diff/diffmatchpatch"
```
Comment (lunny) [inline models/git_diff.go:29]: As above.

File: models/issue_mail.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ package models
 import (
 	"fmt"
 
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline models/issue_mail.go]: external packages should be under internal packages.

File: modules/auth/admin.go [hunk 4]
```diff
@@ -5,9 +5,9 @@
 package auth
 
 import (
-	"gopkg.in/macaron.v1"
+	"gitea.com/macaron/macaron"
```
Comment (lunny) [inline modules/auth/admin.go]: remove the blank line.

File: modules/auth/auth.go [hunk 5]
```diff
@@ -10,11 +10,11 @@ import (
 	"strings"
 	"time"
 
-	"github.com/Unknwon/com"
-	"github.com/go-macaron/binding"
-	"github.com/go-macaron/session"
+	"gitea.com/macaron/binding"
```
Comment (lunny) [inline modules/auth/auth.go]: external packages should be under internal packages.

File: modules/auth/repo_branch_form.go [hunk 6]
```diff
@@ -5,8 +5,8 @@
 package auth
 
 import (
-	"github.com/go-macaron/binding"
-	macaron "gopkg.in/macaron.v1"
+	"gitea.com/macaron/binding"
+	macaron "gitea.com/macaron/macaron"
```
Comment (lunny) [inline modules/auth/repo_branch_form.go]: We should replace all `macaron "gitea.com/macaron/macaron"` to `"gitea.com/macaron/macaron"` because the alias is unnecessary.

File: modules/context/api.go [hunk 7]
```diff
@@ -10,14 +10,14 @@ import (
 	"net/url"
 	"strings"
 
-	"github.com/go-macaron/csrf"
+	"gitea.com/macaron/csrf"
```
Comment (lunny) [inline modules/context/api.go]: Move this external packages under internal packages

File: modules/context/auth.go [hunk 8]
```diff
@@ -10,8 +10,8 @@ import (
 	"code.gitea.io/gitea/modules/auth"
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/setting"
-	"github.com/go-macaron/csrf"
-	macaron "gopkg.in/macaron.v1"
+	"gitea.com/macaron/csrf"
```
Comment (lunny) [inline modules/context/auth.go:14]: blank line.

File: modules/context/context.go [hunk 9]
```diff
@@ -20,12 +20,12 @@ import (
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/setting"
 	"code.gitea.io/gitea/modules/util"
-	"github.com/Unknwon/com"
-	"github.com/go-macaron/cache"
-	"github.com/go-macaron/csrf"
-	"github.com/go-macaron/i18n"
-	"github.com/go-macaron/session"
-	"gopkg.in/macaron.v1"
+	"gitea.com/macaron/cache"
```
Comment (lunny) [inline modules/context/context.go:24]: blank line.

File: modules/context/panic.go [hunk 10]
```diff
@@ -20,7 +20,7 @@ import (
 	"fmt"
 
 	"code.gitea.io/gitea/modules/log"
-	macaron "gopkg.in/macaron.v1"
+	macaron "gitea.com/macaron/macaron"
```
Comment (lunny) [inline modules/context/panic.go]: blank line.

File: modules/options/dynamic.go [hunk 11]
```diff
@@ -12,7 +12,7 @@ import (
 	"path"
 
 	"code.gitea.io/gitea/modules/setting"
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline modules/options/dynamic.go:16]: blank line.

File: modules/options/static.go [hunk 12]
```diff
@@ -12,7 +12,7 @@ import (
 	"path"
 
 	"code.gitea.io/gitea/modules/setting"
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline modules/options/static.go:16]: blank line.

File: modules/public/public.go [hunk 13]
```diff
@@ -14,7 +14,7 @@ import (
 	"time"
 
 	"code.gitea.io/gitea/modules/setting"
-	"gopkg.in/macaron.v1"
+	"gitea.com/macaron/macaron"
```
Comment (lunny) [inline modules/public/public.go:18]: blank line.

File: modules/setting/repository.go [hunk 14]
```diff
@@ -10,7 +10,7 @@ import (
 	"strings"
 
 	"code.gitea.io/gitea/modules/log"
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline modules/setting/repository.go:14]: blank line.

File: modules/templates/dynamic.go [hunk 15]
```diff
@@ -14,8 +14,8 @@ import (
 
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/setting"
-	"github.com/Unknwon/com"
-	"gopkg.in/macaron.v1"
+	"gitea.com/macaron/macaron"
```
Comment (lunny) [inline modules/templates/dynamic.go:18]: blank line.

File: routers/admin/admin.go [hunk 16]
```diff
@@ -13,8 +13,8 @@ import (
 	"strings"
 	"time"
 
-	"github.com/Unknwon/com"
-	"gopkg.in/macaron.v1"
+	"gitea.com/macaron/macaron"
```
Comment (lunny) [inline routers/admin/admin.go]: external packages should be under internal packages.

File: routers/admin/users.go [hunk 17]
```diff
@@ -7,7 +7,7 @@ package admin
 import (
 	"strings"
 
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline routers/admin/users.go]: external packages should be under internal packages.

File: routers/install.go [hunk 18]
```diff
@@ -11,8 +11,8 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/Unknwon/com"
 	"github.com/go-xorm/xorm"
```
Comment (lunny) [inline routers/install.go]: external packages should be under internal packages.

File: routers/org/members.go [hunk 19]
```diff
@@ -5,7 +5,7 @@
 package org
 
 import (
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline routers/org/members.go]: external packages should be under internal packages.

File: routers/org/teams.go [hunk 20]
```diff
@@ -9,7 +9,7 @@ import (
 	"path"
 	"strings"
 
-	"github.com/Unknwon/com"
+	"github.com/unknwon/com"
```
Comment (lunny) [inline routers/org/teams.go]: external packages should be under internal packages.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by tamalsaha** [comment]
@lunny , I took the liberty to update all the Go files. I used [a script](https://github.com/appscodelabs/libbuild/blob/master/reimport3.py) to do this. So, this was easier for me.

**Comment by sapk** [comment]
I have just quickly look at changes and this PR should Aldo fix #7766

**Review by lunny — APPROVED** [review]

File: modules/setting/cors.go [hunk 21]
```diff
@@ -28,7 +28,7 @@ func newCORSService() {
 
 	CORSConfig = cors.Options{
 		Scheme:           sec.Key("SCHEME").String(),
-		AllowDomain:      sec.Key("ALLOW_DOMAIN").String(),
+		AllowDomain:      sec.Key("ALLOW_DOMAIN").Strings(","),
```
Comment (guillep2k) [inline modules/setting/cors.go:31]: This looks unrelated

Comment (tamalsaha) [reply modules/setting/cors.go:31]: This was changed in the gitea fork. https://gitea.com/macaron/cors/commit/609039acef4c33132ea87a69cd69ba826127eff3

Build fails without it.

Comment (guillep2k) [reply modules/setting/cors.go:31]: Oh, I see, it's because Macaron now expects a slice.

Comment (sapk) [reply modules/setting/cors.go:31]: Yes this is related to the issue #7840 that could be closed after this PR.

Comment (HoffmannP) [reply modules/setting/cors.go:31]: @sapk No because point 3. is not yet solved

**Review by guillep2k — APPROVED** [review]
